### ISSUE TITLE
"Cloud Storage" notification and context menu no longer appear in the latest Nightly(04-19-2018)

### ISSUE THREAD

**Issue by SoftVision-EmilPasca** [mozilla/shield-cloudstorage#24]
**[Affected versions]:**
- Latest Nightly 61.0a1-20180419224145 and up

**[Affected Platforms]:**
- All Windows
- All Mac

**[Prerequisites]:**
- The "xpinstall.signatures.required" boolean pref is set to "false".
- The "extensions.legacy.enabled" boolean pref is set tot "true".
- Dropbox Cloud is installed on your coputer: https://www.dropbox.com/install
- Google Drive is installed on your computer: https://www.google.com/drive/download/
- Have the CloudStorage add-on installed.

**[Steps to reproduce]:**
1. Start the Nightly browser with the profile from prerequisites and navigate to: https://www.mozilla.org/en-US/firefox/all/ page.
2. Click any of the "Download" button from the page.
3. Click the "Save" button.
4. Click the download icon from the browser toolbar and observe the Downloads Panel.
5. Right click on a downloaded file and observe the context menu options.

**[Expected result]:**

4. The "Cloud Storage" notification is displayed.
5. The "Cloud Storage" context-menu is displayed.

**[Actual result]:**

4. The "Cloud Storage" notification is not displayed.
5. The "Cloud Storage" context-menu is not available.

**[Regression Window]:**
Last good revision: 3ae765ce2b287848d0dd57292239197f3e3e7988
First bad revision: 74478b16e546d7c5c9c77552efb5f16a49811502
Pushlog:
https://hg.mozilla.org/integration/autoland/pushloghtml?fromchange=3ae765ce2b287848d0dd57292239197f3e3e7988&tochange=74478b16e546d7c5c9c77552efb5f16a49811502

**[Notes]:**
- The issue is not reproducible with Nightly(61.0a1-20180418230818).
- Attached a screen recording of the issue:
![cloudstorage_doesntwork](https://user-images.githubusercontent.com/20083658/39056698-18586ac2-44c0-11e8-8a24-6c51db26df97.gif)

**Comment by punamdahiya** [comment]
Thanks @SoftVision-EmilPasca for logging this issue. With landing of https://bugzilla.mozilla.org/show_bug.cgi?id=1034036 , resource:///modules/RecentWindow.jsm module is no more available in 61 nightly.

61+, It's been replaced by resource:///modules/BrowserWindowTracker.jsm. I can do a version of extension for 61 as well, however since shield study is planned for 60, we should definitely continue testing in 60 beta  and logs issue against it. Thanks!

**Comment by punamdahiya** [comment]
Just realized till we have a signed extension, we can test in 60 developer edition. Thanks!

**Comment by punamdahiya** [comment]
[cloud_storage_shield_study_web_extension_experiment_cssswee_-1.0.0.zip](https://github.com/mozilla/shield-cloudstorage/files/1973159/cloud_storage_shield_study_web_extension_experiment_cssswee_-1.0.0.zip)
Updating extension that uses "resource:///modules/BrowserWindowTracker.jsm" module for extension to work in nightly

**Comment by punamdahiya** [comment]
@aswan failed to install above extension in nightly. The error seen in console on debug of the extension is
TypeError: "browser.cloudstorage is undefined"
<anonymous>moz-extension://39365f63-bb49-ef47-acb2-620a40790af7/background.js:2:3

The only code change in extension thats working on Firefox 60 is using module BrowserWindowTracker.jsm and using BrowserWindowTracker.getTopWindow(); here

https://github.com/mozilla/shield-cloudstorage/blob/cloud-storage-web-ext/webextension/privileged/ui/CloudDownloadsView.jsm#L37

Does WEE needs some other settings to run in nightly that are missed in above extension?

**Comment by punamdahiya** [comment]
Confirmed with @aswan over irc, for attached extension to work in nightly we need to set two prefs:
extension.legacy.enabled to true and xpinstall.signature.required to false

**Comment by punamdahiya** [comment]
I see QA has it listed in pre-requisite, missed it while testing in nightly

**Comment by punamdahiya** [comment]
@SoftVision-EmilPasca @SoftVision-CosminMuntean I have merged  cloud-storage-web-ext to master, please use master branch for testing forward. 

Master should have #24 fixed, please validate. Thanks!

**Comment by SoftVision-CosminMuntean** [comment]
I have retested this issue on latest Nightly (61.0a1) build with latest Cloud Storage version from master branch and the issue is no longer reproducible. The extensions is correctly installed on latest Nightly builds and the Cloud Storage notification is displayed.
Tested on Windows 10 x64, Windows 7 x64, Mac 10.13 and Ubuntu 16.0.4 x64.

I have also tried on latest Firefox (59.0.3) release and it seems that the Cloud Storage notification is not displayed. In order to install Cloud Storage on Firefox 59.0.3 I have download an [Unbranded build](https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds).
However, since Cloud Storage shield study is targeted for Firefox 60, I think this issue doesn't affects us.
